### PR TITLE
Doxygen doc overhaul

### PR DESCRIPTION
**Pull request by only-a-ptr**
I updated the looks (doxygen-awesome) and placed my overwiew text on the main page.

image

### CODE REVIEW

**Comment by Zentro** [comment]
Is there somewhere published that this can be viewed? @only-a-ptr

**Comment by AnotherFoxGuy** [comment]
> @Zentro  Is there somewhere published that this can be viewed?

Here is a preview: https://test.anotherfoxguy.com/

**Comment by only-a-ptr** [comment]
@AnotherFoxGuy thanks!
@Zentro I'd like you to publish it as http://developer.rigsofrods.org/, ideally with versioning like OGRE does: /latest for latest release, /VERSION for previous releases and as an added feat, /nightly for current master branch.

File: doc/doxygen/MainPage.h [hunk 1]
```diff
@@ -1,88 +1,200 @@
...
- * 
- *  
- * 
- * ## License of Rigs of Rods
- * 
  * > Copyright (c) 2005-2013 Pierre-Michel Ricordel  
- * > Copyright (c) 2007-2013 Thomas Fischer  
- * > Copyright (c) 2009-2015 Rigs of Rods Contributors
+ * > Copyright (c) 2007-2013 Thomas Fischer
+ * > Copyright (c) 2013-2022 Thomas Fischer
```
Comment (Zentro) [inline doc/doxygen/MainPage.h:196]: Thomas is shown twice here. Should be Copyright (c) 2013-2022 Petr Ohlidal (& contributors).

**Review by Zentro — CHANGES_REQUESTED** [review]

**Comment by Zentro** [comment]
> @AnotherFoxGuy thanks! @Zentro I'd like you to publish it as http://developer.rigsofrods.org/, ideally with versioning like [OGRE does](https://ogrecave.github.io/ogre/api/latest/): /latest for latest release, /VERSION for previous releases and as an added feat, /nightly for current master branch.

Will do.

**Comment by only-a-ptr** [comment]
Guys, I need your opinions - does this make sense? https://github.com/RigsOfRods/rigs-of-rods/pull/2659/commits/cf12bafb09c8e06fb288db9832387bf9b5ed9cc8

I'm trying to document 4 different things in a single doxygen doc:
* the codebase
* angelscript API exposed by game to scripts (Script2Game)
* angelscript callbacks which the game finds and invokes in scripts (Game2Script)
* angelscript features from builtin scripts - currently it's just raceManager, but there will be more.

Is it clear what is what?

**Comment by tritonas00** [comment]
makes sense

**Comment by Zentro** [comment]
Makes sense for me too.

**Review by Zentro — APPROVED** [review]

**Review by tritonas00 — APPROVED** [review]

**Pull request merged**